common/unsafe/src/main/java/org/apache/spark/unsafe/types/UTF8String.java

```diff
@@ -952,6 +952,11 @@ public static UTF8String concatWs(UTF8String separator, UTF8String... inputs) {
   }
 
   public UTF8String[] split(UTF8String pattern, int limit) {
+    // Java String's split method supports "ignore empty string" behavior when the limit is 0.
+    // To avoid this, we fall back to -1 when the limit is 0.
```

Comment:
I also would leave a short justification for this given https://github.com/apache/spark/pull/22227#issuecomment-417471241